src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/Utils/TypeNameHelpers.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+using System.Runtime.CompilerServices;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace System.Formats.Nrbf.Utils;
+
+internal static class TypeNameHelpers
+{
+    // PrimitiveType does not define Object, IntPtr or UIntPtr
+    internal const PrimitiveType ObjectPrimitiveType = (PrimitiveType)19;
+    internal const PrimitiveType IntPtrPrimitiveType = (PrimitiveType)20;
+    internal const PrimitiveType UIntPtrPrimitiveType = (PrimitiveType)21;
+    private static readonly TypeName?[] s_PrimitiveTypeNames = new TypeName?[(int)UIntPtrPrimitiveType + 1];
+    private static readonly TypeName?[] s_PrimitiveSZArrayTypeNames = new TypeName?[(int)UIntPtrPrimitiveType + 1];
+    private static AssemblyNameInfo? s_CoreLibAssemblyName;
```

Comment:
Incorrect casing in these names

```suggestion
    private static readonly TypeName?[] s_primitiveTypeNames = new TypeName?[(int)UIntPtrPrimitiveType + 1];
    private static readonly TypeName?[] s_primitiveSZArrayTypeNames = new TypeName?[(int)UIntPtrPrimitiveType + 1];
    private static AssemblyNameInfo? s_coreLibAssemblyName;
```